packages/ses/src/shim-arraybuffer-transfer.js

```diff
@@ -0,0 +1,61 @@
+import {
+  arrayBufferPrototype,
+  arrayBufferSlice,
+  globalThis,
+  TypeError,
+  defineProperty,
+} from './commons.js';
+
+export const shimArrayBufferTransfer = () => {
+  // @ts-expect-error TODO extend ArrayBuffer type to include transfer, etc.
+  if (typeof arrayBufferPrototype.transfer === 'function') {
+    // Assume already exists so does not need to be shimmed.
+    // Such conditional shimming is ok in this case since ArrayBuffer.p.transfer
+    // is already officially part of JS.
+    return;
+  }
+  const clone = globalThis.structuredClone;
+  if (typeof clone !== 'function') {
+    // Indeed, Node <= 16 has neither.
+    throw TypeError(
+      `Can only shim missing ArrayBuffer.prototype.transfer on a platform with "structuredClone"`,
+    );
+  }
+
+  /**
+   * @type {ThisType<ArrayBuffer>}
+   */
+  const methods = {
+    /**
+     * @param {number} [newLength]
+     */
+    transfer(newLength = undefined) {
+      // Hopefully, a zero-length slice is cheap, but still enforces that
+      // `this` is a genuine `ArrayBuffer` exotic object.
+      arrayBufferSlice(this, 0, 0);
+      const oldLength = this.byteLength;
```

Comment:
Why not use the [getter in `ArrayBuffer.prototype.byteLength`](https://tc39.es/ecma262/#sec-get-arraybuffer.prototype.bytelength) to do both the brand check and get the value here if we don't trust the `this` object?

```suggestion
      const oldLength = arrayBufferGetByteLength(this);
```

Reply:
Good idea! Done.